### PR TITLE
feat(queries): update idl hightlight

### PR DESCRIPTION
**Pull request by cathaysia**
here is a idl file for test new queries:

```idl
home a manages B {};

home A: B manages B {};

home A: B manages B {
    void func1();
    void func1() raises(A);
    void func1() raises(A,B,C);
    void func1(in u8 attr, out u16 attr);
    void func1(in u8 attr, out u16 attr) raises(A);
    void func1(in u8 attr, out u16 attr) raises(A,B,C);
    readonly attribute u8 attr1, attr2, attr3;
    readonly attribute u8 attr1 raises(A);
    readonly attribute u8 attr1 raises(A, B, C);
    attribute u8 attr1, attr2, attr3;
    attribute u8 attr1 getraises(A);
    attribute u8 attr1 getraises(A, B, C) setraises(A);
    attribute u8 attr1 setraises(A);

    factory A ();
    factory A (in B a);
    factory A (in B a) raises(A);
    factory A (in B a) raises(A,B,C);
};


typeid A::A "bbb";
typeprefix A "ccc";
import "aaaa";
import A::A;

local interface A {
    attribute Object attr1;
    oneway void aaa (in u8 a);
    oneway void aaa (in u8 a) context();
    oneway void aaa (in u8 a) context("aaa", "bbb");
};

custom valuetype HelloWorld {};
abstract valuetype HelloWorld {};
abstract valuetype HelloWorld : A {};

local interface A {
    attribute Object attr1;
    attribute ValueBase attr1;
};

custom valuetype A: B {
};

custom valuetype A supports B {};

custom valuetype A: B supports C {};

valuetype HelloWorld; // forward declare
valuetype A a; // value_box_def
valuetype A sequence<string>; // value_box_def
custom valuetype A a; // value_box_def
// ^ extend grammar

component A;

component A: B {
    provides A a;
    uses A b;
    readonly attribute u8 attr1, attr2, attr3;
    readonly attribute u8 attr1 raises(A);
    readonly attribute u8 attr1 raises(A, B, C);
    attribute u8 attr1, attr2, attr3;
    attribute u8 attr1 getraises(A);
    attribute u8 attr1 getraises(A, B, C) setraises(A);
    attribute u8 attr1 setraises(A);
};

home a manages B {};

home A: B manages B {};

home A: B manages B {
    void func1();
    void func1() raises(A);
    void func1() raises(A,B,C);
    void func1(in u8 attr, out u16 attr);
    void func1(in u8 attr, out u16 attr) raises(A);
    void func1(in u8 attr, out u16 attr) raises(A,B,C);
    readonly attribute u8 attr1, attr2, attr3;
    readonly attribute u8 attr1 raises(A);
    readonly attribute u8 attr1 raises(A, B, C);
    attribute u8 attr1, attr2, attr3;
    attribute u8 attr1 getraises(A);
    attribute u8 attr1 getraises(A, B, C) setraises(A);
    attribute u8 attr1 setraises(A);

    factory A ();
    factory A (in B a);
    factory A (in B a) raises(A);
    factory A (in B a) raises(A,B,C);
};

porttype A;

porttype A {
    port u8 a;
};

connector A: B {
    port u8 a;
};

porttype ControlledConsumer {
    provides Pusher consumer;
    uses FlowControl control;
};

connector Cnx {
    mirrorport ControlledConsumer cc;
    provides Pusher p;
};

component A: B supports A, B, C{
    provides Object a;
    provides A a;
    uses A b;
    readonly attribute u8 attr1, attr2, attr3;
    readonly attribute u8 attr1 raises(A);
    readonly attribute u8 attr1 raises(A, B, C);
    attribute u8 attr1, attr2, attr3;
    attribute u8 attr1 getraises(A);
    attribute u8 attr1 getraises(A, B, C) setraises(A);
    attribute u8 attr1 setraises(A);
    emits A b;
    publishes A b;
    consumes A b;
};

home A: B manages B primarykey B {
    finder aa(in u8 a) raises(A,B,C);

};

abstract eventtype A;

custom eventtype A: B {
    readonly attribute u8 attr1, attr2, attr3;
    readonly attribute u8 attr1 raises(A);
    readonly attribute u8 attr1 raises(A, B, C);
    attribute u8 attr1, attr2, attr3;
    attribute u8 attr1 getraises(A);
    attribute u8 attr1 getraises(A, B, C) setraises(A);
    attribute u8 attr1 setraises(A);
};

abstract eventtype A: B {
    readonly attribute u8 attr1, attr2, attr3;
    readonly attribute u8 attr1 raises(A);
    readonly attribute u8 attr1 raises(A, B, C);
    attribute u8 attr1, attr2, attr3;
    attribute u8 attr1 getraises(A);
    attribute u8 attr1 getraises(A, B, C) setraises(A);
    attribute u8 attr1 setraises(A);
};
```

### CODE REVIEW

**Comment by cathaysia** [comment]
there is a test not pass, but it maybe not my fault.

**Review by ribru17 — APPROVED** [review]

**Comment by ribru17** [comment]
I've noticed the `@variable.member` highlight is kinda overly generous (taking precedence over `@variable.parameter`). Is there an easy way to fix that?

File: queries/idl/highlights.scm [hunk 1]
```diff
@@ -8,19 +8,44 @@
...
+  (value_kind)
+  "component"
+  "porttype"
+  "connector"
+  "eventtype"
+  "valuetype"
 ] @keyword.type
 
+(import_dcl
+  "import" @keyword.directive)
```
Comment (lucario387) [inline queries/idl/highlights.scm:26]: Shouldn't this be `@keyword.import`, looking like it's part of import declaration

File: queries/idl/highlights.scm [hunk 2]
```diff
@@ -203,3 +231,106 @@
...
+
+(port_dcl
+  (scoped_name
+    (identifier))
+  (identifier) @variable.member)
+
+(port_dcl
+  (scoped_name
+    (identifier))
+  (identifier) @variable.member)
```
Comment (lucario387) [inline queries/idl/highlights.scm:302]: Duplicates

Comment (lucario387) [reply queries/idl/highlights.scm]: Still duplicates :D

Comment (cathaysia) [reply queries/idl/highlights.scm]: It should be fine now

**Comment by cathaysia** [comment]
> I've noticed the `@variable.member` highlight is kinda overly generous (taking precedence over `@variable.parameter`). Is there an easy way to fix that?

I reviewed the newly added code. However, since most of the new syntax is part of the type declaration, the number of @variable.member(and @type) is very large. @variable.paramer should apply to function parameters or template parameters. But these are very few in the idl syntax.

File: queries/idl/highlights.scm [hunk 3]
```diff
@@ -8,19 +8,43 @@
...
+  "valuetype"
 ] @keyword.type
 
+"import" @keyword.import
+
 [
   "module"
   "attribute"
+  "factory"
+  "manages"
```
Comment (ribru17) [inline queries/idl/highlights.scm:31]: Can you add `"alias"` please

**Comment by ribru17** [comment]
> I reviewed the newly added code. However, since most of the new syntax is part of the type declaration, the number of @variable.member(and @type) is very large. @variable.paramer should apply to function parameters or template parameters. But these are very few in the idl syntax.

I see: is it possible to do this for just signature declarations then? Something like
```idl
void on_creation (in T datum, in ReadInfo info);
```
has `datum` and `info` overridden by `@attribute`

**Comment by cathaysia** [comment]
Yes, because I didn't inject highlighting into it, causing it to use a more generic highlight

**Comment by ribru17** [comment]
I think that shouldn't be necessary, if you can just change:
```query
(param_dcl
  (simple_declarator) @variable.parameter)
```
to
```query
(param_dcl
  (simple_declarator
    (identifier) @variable.parameter))
```
and then move it further down the query list then it will at least take precedence over the `@attribute` highlight.

**Comment by cathaysia** [comment]
I removed the overly general rules and added new ones. Now @variable.attribute in op_dcl and param_dcl should be highlighted normally

File: queries/idl/highlights.scm [hunk 4]
```diff
@@ -195,11 +218,104 @@
...
+  (scoped_name)
+  (identifier) @variable.member)
+
+(publishes_dcl
+  (scoped_name)
+  (identifier) @variable.member)
+
+(consumes_dcl
+  (scoped_name)
+  (identifier) @variable.member)
```
Comment (lucario387) [inline queries/idl/highlights.scm:305]: Reading through the parser's code, I believe the `scoped_name` here can be removed without issues. emits_dcl is a sequence of `emits -> (scoped_name) -> (identifier)` so `(emits_dcl (identifier))` should be equivalent to `(emits_dcl (scoped_name) (identifier))`. Same goes for other places.

Comment (cathaysia) [reply queries/idl/highlights.scm]: yes, you are right. I removed the useless codes. :)

File: queries/idl/highlights.scm [hunk 5]
```diff
@@ -203,3 +233,100 @@
...
+  (identifier) @type)
+
+(porttype_def
+  (identifier) @type)
+
+(port_dcl
+  (identifier) @variable.member)
+
+(port_dcl
+  (identifier) @variable.member)
```
Comment (lucario387) [inline queries/idl/highlights.scm:297]: This is still duplicated

Comment (cathaysia) [reply queries/idl/highlights.scm]: It should be fixed now

**Review by lucario387 — APPROVED** [review]

**Comment by cathaysia** [comment]
can this be merged?

**Pull request merged**